packages/types/types/src/record.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IConsumer<T> {
+    /**
+     * Invoked whenever the data this object is bound to is changed.
+     */
+    valueChanged<U extends T, K extends keyof U>(property: K, producer: IProducer<U>): void;
+}
+
+export interface IReader<T> {
+    /**
+     * Return the value associated with `property`.
+     * @param property - The property of the Producer to read.
+     */
+    get<K extends keyof T>(property: K): T[K];
+
+    /**
+     * A reference to the underlying producer that provides values for this reader.
+     */
+    readonly producer?: IProducer<T>;
+}
+
+export interface IWriter<T> {
+    set<K extends keyof T>(property: K, value: T[K]): void;
```

Comment:
I think it would be helpful to standardise clearing keys across the other types too. (A later task). I think there is a grey area around what setting undefined has meant to the matrix.

Reply:
Good point.  I'll open an issue.